Clean up generation of table report

johanneskoester opened the pull request:
Further, modify the config file to list the protein in addition to the VOCs.

thomasbtf commented on .github/workflows/main.yml:
```diff
@@ -65,8 +65,10 @@ jobs:
+      - name: Install dependcies
+        run: |
+          sudo apt-get install tree
+          sudo apt-get install bc
```
Apparently, `bc` is not installed on the GitHub Action Runners. I think the more proper way would be to use other tools than `bc` to keep the workflow portable, right?

thomasbtf replied:
Removed `bc` in b96ca21

johanneskoester replied:
bc should be added to the conda env. But you have found a bash only solution?

thomasbtf replied:
Yes - instead of piping to bc, I wrapped `zcat {input.fastq1} | wc -l) / 4` in an arithmetic expression `$((...))`. 

There are of course other options if arithmetic expressions are not appropriate: https://www.shell-tips.com/bash/math-arithmetic-calculation/

thomasbtf commented:
It seems, that I broke the formatting by the `report-table-formatter.js` when I renamed the column headers of the table. Should I revert the renaming or can we "easily" adjust the 'report-table-formatter.js'?

Moreover, I noticed that the column `Variants of Interest` contains 2x `S:N501Y`. Maybe the set operation [here](https://github.com/koesterlab/snakemake-workflow-sars-cov2/pull/144/files#diff-e8e490d664ca82d902d5c8190b06a1b4b12de0317c04a8ce8ae9e721db78be6fR168) does not suffice, because the Allele frequency is included in the string?


![image](https://user-images.githubusercontent.com/46334240/116658050-ae9eeb00-a98f-11eb-9ad8-a745be18e878.png)

johanneskoester commented on workflow/scripts/generate-overview-table.py:
```diff
@@ -170,4 +171,11 @@ def register_contig_lengths(assemblies, name):
+    str : ["pangolin strain (#SNPs)", "variants of interest", "other variants"]
+}
+for dtype, columns in fmt_dict.items(): 
+    data[columns] = data[columns].astype(dtype)
```
Are you sure this is needed? If it is needed, it means to me that we insert the values in the wrong datatypes.

thomasbtf replied:
Ahh correct, I didn't think enough, when I did that. Removed the `str` casting in 55e5c15.

johanneskoester commented:
> It seems, that I broke the formatting by the `report-table-formatter.js` when I renamed the column headers of the table in
> [0242fc3](https://github.com/koesterlab/snakemake-workflow-sars-cov2/commit/0242fc3d0c9432786e639c2a16b57b7ab6bce1b5). Should I revert the renaming or can we "easily" adjust the 'report-table-formatter.js'?
> 
> Moreover, I noticed that the column `Variants of Interest` contains 2x `S:N501Y`. Maybe the set operation [here](https://github.com/koesterlab/snakemake-workflow-sars-cov2/pull/144/files#diff-e8e490d664ca82d902d5c8190b06a1b4b12de0317c04a8ce8ae9e721db78be6fR168) does not suffice, because the Allele frequency is included in the string?
> 
> ![image](https://user-images.githubusercontent.com/46334240/116658050-ae9eeb00-a98f-11eb-9ad8-a745be18e878.png)

Ok I have added a commit that fixes the duplication issue.
For the column names, I like your changes. All you need to do is to adjust the names here: https://github.com/koesterlab/snakemake-workflow-sars-cov2/blob/master/resources/report-table-formatter.js#L95

alethomas commented:
is there a reason this isn't merged yet?

thomasbtf commented:
We still have to include a linebreak, if 2 vocs are identified
